Marlin/src/module/stepper.h

```diff
@@ -202,8 +210,12 @@
   #error "Expected at least one of MINIMUM_STEPPER_PULSE or MAXIMUM_STEPPER_RATE to be defined"
 #endif
 
-// But the user could be enforcing a minimum time, so the loop time is
-#define ISR_LOOP_CYCLES (ISR_LOOP_BASE_CYCLES + _MAX(MIN_STEPPER_PULSE_CYCLES, MIN_ISR_LOOP_CYCLES))
+// The loop takes the base time plus the time for all the bresenham logic for R pulses plus the time
+// between pulses for (R-1) pulses. But the user could be enforcing a minimum time so the loop time is:
+#define ISR_LOOP_CYCLES(R) ((ISR_LOOP_BASE_CYCLES + MIN_ISR_LOOP_CYCLES + MIN_STEPPER_PULSE_CYCLES) * (R - 1) + _MAX(MIN_ISR_LOOP_CYCLES, MIN_STEPPER_PULSE_CYCLES))
```

Comment:
`ISR_LOOP_CYCLES(R)` revised as below fixes skipping stepper motor driver (A4988) on AVR with tested with input shaping off and on. 
```
#define ISR_LOOP_CYCLES(R) ((ISR_LOOP_BASE_CYCLES) * (R - 1) + _MAX(MIN_ISR_LOOP_CYCLES, MIN_STEPPER_PULSE_CYCLES))
```
There is still a smaller "skip" in the motor at high speeds when input shaping is turned on that I haven't been able to figure out yet. Is there a reason that `#define ISR_BASE_CYCLES 1000UL` was changed from `752UL`?

Reply:
I'm going to reply in #25117 to keep the conversation in once place.